libc/sock/internal.h

```diff
@@ -53,6 +53,45 @@ struct sockaddr_un_bsd {
   char sun_path[108];
 };
 
+/* -----------------------------------------------------------------------------------
+ * ioctl SIOCGIFCONF & others:
+ *
+ * BSD has a different structure. All the ioctl will adjust to use
+ * the ifreq with the ifreq_bsd
+ */
+struct ifreq_bsd {
+  union {
+    char ifrn_name[IFNAMSIZ];
+  } ifr_ifrn;
+
+  union {
+    /* Right now we only list the structures used by the few ioctl that are
+     * supported by Cosmopolitan. Add your own definition here if you need
+     * a particular ioctl that requires polyfill for BSD
+     */
+    struct  sockaddr_bsd ifru_addr;
+    struct  sockaddr_bsd ifru_netmask;
+    struct  sockaddr_bsd ifru_dstaddr;
+    struct  sockaddr_bsd ifru_broadaddr;
+    short   ifru_flags;
+    char    ifru_pad[16];   /* used as padding */
+  } ifr_ifru;
+};
+
+#pragma pack(4)
+struct ifconf_bsd {
```

Comment:
I hadn't heard of pragma pack before. Other ways of doing it are `_Alignas(4)` on the field or `forcealign(4)`. It's also worth noting that XNU is the only BSD-like system that has this peculiarity.  I'll see if I can iterate on this locally since I've got a bunch of VMs set up.